swarm.go

```diff
@@ -178,6 +182,10 @@ func (s *Swarm) Process() goprocess.Process {
 }
 
 func (s *Swarm) addConn(tc transport.CapableConn, dir network.Direction) (*Conn, error) {
+	numOfPeers := len(s.conns.m)
+	if numOfPeers >= int(s.peerLimit) && s.Connectedness(tc.RemotePeer()) != network.Connected && s.peerLimit != 0 {
+		return nil, errors.New("number of peers over the peer limit, so rejecting connection")
+	}
```

Comment:
Two things wrong with this:
- you need a default of 0 to mean no peer limit, so that thing work as they currently do (this should be the default)
- you are leaking the connection here, you need to close it if you are rejecting.

Reply:
Hey the if the peer limit is set to zero it ignores this conditional. 
```go 
&& s.peerLimit != 0
```
 I added a var to make this more clear. 
I also now close the connection in the event of an error. 

If I want to set a fixed peer limit , after which no peer will be able to connect with us what would be the best way to go about this ?

Reply:
Put the peer limit not being 0 first please, this was hard to parse.
Also, it's the default, so it will be faster to short-circuit.

In addition, you should log something about this connection closure.